Remove WTF::allOf from the codebase.

c8ef opened the pull request:
#### 3126ee91ff79ce76f1dce73e23b68fdb93484c13
<pre>
Remove WTF::allOf from the codebase.
<a href="https://bugs.webkit.org/show_bug.cgi?id=286659">https://bugs.webkit.org/show_bug.cgi?id=286659</a>

Reviewed by NOBODY (OOPS!).

This API is not necessary with the newer version of STL.

* Source/WTF/wtf/Algorithms.h:
(WTF::allOf): Deleted.
* Source/WebCore/Modules/airplay/WebMediaSessionManager.cpp:
(WebCore::WebMediaSessionManager::alwaysOnLoggingAllowed const):
* Source/WebCore/Modules/mediastream/RTCPeerConnection.cpp:
(WebCore::isIceTransportUsedByTransceiver):
(WebCore::RTCPeerConnection::computeConnectionState):
(WebCore::RTCPeerConnection::computeIceConnectionStateFromIceTransports):
* Source/WebCore/Modules/mediastream/UserMediaRequest.cpp:
(WebCore::isMediaStreamCorrectlyStarted):
* Source/WebCore/Modules/mediastream/gstreamer/GStreamerMediaEndpoint.cpp:
(WebCore::GStreamerMediaEndpoint::createTransceiverBackends):
* Source/WebCore/css/typedom/CSSNumericValue.cpp:
(WebCore::operationOnValuesOfSameUnit):
(WebCore::CSSNumericValue::multiplyInternal):
(WebCore::CSSNumericValue::equals):
* Source/WebCore/css/typedom/transform/CSSTransformValue.cpp:
(WebCore::CSSTransformValue::is2D const):
* Source/WebCore/dom/Document.cpp:
(WebCore::Document::configureSharedLogger):
* Source/WebCore/html/CanvasNoiseInjection.cpp:
(WebCore::boundingNeighbors):
* Source/WebCore/page/csp/ContentSecurityPolicy.cpp:
(WebCore::ContentSecurityPolicy::allowMissingTrustedTypesForSinkGroup const):
* Source/WebCore/platform/graphics/ContentTypeUtilities.cpp:
(WebCore::contentTypeMeetsContainerAndCodecTypeRequirements):
* Source/WebCore/platform/graphics/avfoundation/CDMFairPlayStreaming.cpp:
(WebCore::CDMPrivateFairPlayStreaming::supportsConfigurationWithRestrictions const):
* Source/WebKit/GPUProcess/mac/GPUProcessMac.mm:
(WebKit::GPUProcess::setScreenProperties):
* Source/WebKit/NetworkProcess/ServiceWorker/WebSWServerConnection.cpp:
(WebKit::WebSWServerConnection::resolveUnregistrationJobInClient):
* Source/WebKit/NetworkProcess/storage/NetworkStorageManager.cpp:
(WebKit::isEmptyOriginDirectory):
* Source/WebKit/UIProcess/WebProcessProxy.cpp:
(WebKit::WebProcessProxy::isAlwaysOnLoggingAllowed const):
* Source/WebKit/WebProcess/WebProcess.cpp:
(WebKit::WebProcess::areAllPagesThrottleable const):
* Source/WebKit/WebProcess/cocoa/WebProcessCocoa.mm:
(WebKit::WebProcess::updatePageScreenProperties):
* Tools/TestWebKitAPI/Tests/WTF/WeakPtr.cpp:
(TestWebKitAPI::TEST(WTF_WeakPtr, WeakHashMapIterators)):
</pre><!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/3126ee91ff79ce76f1dce73e23b68fdb93484c13

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/87732 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/7248 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/42117 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/92597 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/38481 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/89783 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/7629 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/15419 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/67736 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/25484 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/90734 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/5822 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/79378 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/48107 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/5609 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/33780 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/37589 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/80530 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/76007 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/34663 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/94483 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/86507 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/14900 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/10937 "Found 2 new test failures: ipc/create-media-source-with-invalid-constraints-crash.html media/video-replaces-poster.html (failure)") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/76589 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/15154 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/75234 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/75824 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/20185 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/18617 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/7874 "The change is no longer eligible for processing.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/14916 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/20219 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/109001 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/14660 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/26209 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/18104 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/16442 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/add2817c2b7fb73c8622d32c0a61e130960c000d. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/39668

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/d1e0ac205993b7ab9c4ad07733eceb3c74408619. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/39668

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/dfa52f6b3e461429284fde044bf7afa5856b67c5. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/39668

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/78e659f8075609904a8b43626c5498c697932f9c. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/39668

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/518c6d3a4e826caca3565c50cf47a78af35dd966. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/39668

geoffreygaren commented on Source/WebCore/css/typedom/CSSNumericValue.cpp:
```diff
@@ -333,8 +333,8 @@ bool CSSNumericValue::equals(FixedVector<CSSNumberish>&& values)
     auto numericValues = WTF::map(WTFMove(values), rectifyNumberish);
-    return WTF::allOf(numericValues, [&] (const Ref<CSSNumericValue>& value) {
-        return this->equals(value.get());
+    return std::ranges::all_of(numericValues, [protectedThis = Ref { *this }] (const Ref<CSSNumericValue>& value) {
+        return protectedThis->equals(value.get());
```
WTF::allOf includes a `NOESCAPE` annotation that eliminates the need for this refcount. Style improvements are nice, but only when they achieve equal functionality.

geoffreygaren replied:
But... why? Sure, you can replace uses of allOf with loops, but the purpose of allOf is not to have to write out a loop.

If std::ranges::all_of is less capable than WTF::allOf, then we should not remove WTF::allOf.

c8ef committed:
Remove WTF::allOf from the codebase.
https://bugs.webkit.org/show_bug.cgi?id=286659

Reviewed by NOBODY (OOPS!).

This API is not necessary with the newer version of STL.

* Source/WTF/wtf/Algorithms.h:
(WTF::allOf): Deleted.
* Source/WebCore/Modules/airplay/WebMediaSessionManager.cpp:
(WebCore::WebMediaSessionManager::alwaysOnLoggingAllowed const):
* Source/WebCore/Modules/mediastream/RTCPeerConnection.cpp:
(WebCore::isIceTransportUsedByTransceiver):
(WebCore::RTCPeerConnection::computeConnectionState):
(WebCore::RTCPeerConnection::computeIceConnectionStateFromIceTransports):
* Source/WebCore/Modules/mediastream/UserMediaRequest.cpp:
(WebCore::isMediaStreamCorrectlyStarted):
* Source/WebCore/Modules/mediastream/gstreamer/GStreamerMediaEndpoint.cpp:
(WebCore::GStreamerMediaEndpoint::createTransceiverBackends):
* Source/WebCore/css/typedom/CSSNumericValue.cpp:
(WebCore::operationOnValuesOfSameUnit):
(WebCore::CSSNumericValue::multiplyInternal):
(WebCore::CSSNumericValue::equals):
* Source/WebCore/css/typedom/transform/CSSTransformValue.cpp:
(WebCore::CSSTransformValue::is2D const):
* Source/WebCore/dom/Document.cpp:
(WebCore::Document::configureSharedLogger):
* Source/WebCore/html/CanvasNoiseInjection.cpp:
(WebCore::boundingNeighbors):
* Source/WebCore/page/csp/ContentSecurityPolicy.cpp:
(WebCore::ContentSecurityPolicy::allowMissingTrustedTypesForSinkGroup const):
* Source/WebCore/platform/graphics/ContentTypeUtilities.cpp:
(WebCore::contentTypeMeetsContainerAndCodecTypeRequirements):
* Source/WebCore/platform/graphics/avfoundation/CDMFairPlayStreaming.cpp:
(WebCore::CDMPrivateFairPlayStreaming::supportsConfigurationWithRestrictions const):
* Source/WebKit/GPUProcess/mac/GPUProcessMac.mm:
(WebKit::GPUProcess::setScreenProperties):
* Source/WebKit/NetworkProcess/ServiceWorker/WebSWServerConnection.cpp:
(WebKit::WebSWServerConnection::resolveUnregistrationJobInClient):
* Source/WebKit/NetworkProcess/storage/NetworkStorageManager.cpp:
(WebKit::isEmptyOriginDirectory):
* Source/WebKit/UIProcess/WebProcessProxy.cpp:
(WebKit::WebProcessProxy::isAlwaysOnLoggingAllowed const):
* Source/WebKit/WebProcess/WebProcess.cpp:
(WebKit::WebProcess::areAllPagesThrottleable const):
* Source/WebKit/WebProcess/cocoa/WebProcessCocoa.mm:
(WebKit::WebProcess::updatePageScreenProperties):
* Tools/TestWebKitAPI/Tests/WTF/WeakPtr.cpp:
(TestWebKitAPI::TEST(WTF_WeakPtr, WeakHashMapIterators)):

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/3126ee91ff79ce76f1dce73e23b68fdb93484c13. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/39668

geoffreygaren reviewed (changes requested):
I don't think we should do this.
